datalad/distribution/update.py

```diff
@@ -226,6 +233,13 @@ def __call__(
             reobtain_data=False):
         if fetch_all is not None:
             lgr.warning('update(fetch_all=...) called. Option has no effect, and will be removed')
+        if sibling is not None:
```

Comment:
ideally, for those deprecated we should use some dedicated (not just `None`) singleton... in #5517 I introduced NoneDeprecated, check it out.  Then even if someone used explicitly `sibling=None` for some obnoxious reason, we would "detect" that

Reply:
The `NoneDeprecated` type is now actually merged and available.